lib/dao.js

```diff
@@ -1646,22 +1646,32 @@ function coerceArray(val) {
 
 DataAccessObject._getHiddenProperties = function() {
   var settings = this.definition.settings || {};
-  var result = settings.hiddenProperties || settings.hidden;
+  var result = settings.hiddenProperties || settings.hidden || [];
   if (typeof result === 'string') {
     result = [result];
   }
-  result = result || [];
-  return result;
+  if (Array.isArray(result)) {
+    return result;
+  } else {
+    // See https://github.com/strongloop/loopback-datasource-juggler/issues/1646
+    // `ModelBaseClass` normalize the properties to an object such as `{secret: true}`
+    return Object.keys(result);
```

Comment:
I think doing a straight `Object.keys` is a bit misleading without doing a bit of filtering. 

The way it stands it seems to me `{secret: false}` would still be filtered as hidden.